modin/experimental/core/execution/native/implementations/hdk_on_native/partitioning/partition_manager.py

```diff
@@ -248,11 +248,21 @@ def run_exec_plan(cls, plan):
         for frame in frames:
             cls.import_table(frame, worker)
 
-        calcite_plan = CalciteBuilder().build(plan)
+        builder = CalciteBuilder()
+        calcite_plan = builder.build(plan)
         calcite_json = CalciteSerializer().serialize(calcite_plan)
         if DoUseCalcite.get():
+            exec_calcite = True
             calcite_json = "execute calcite " + calcite_json
-        table = worker.executeRA(calcite_json)
+        else:
+            exec_calcite = False
+        if builder.has_groupby == builder.has_join:
+            exec_args = {}
+        elif builder.has_groupby:
+            exec_args = {"enable_lazy_fetch": 0, "enable_columnar_output": 0}
+        elif builder.has_join:
+            exec_args = {"enable_lazy_fetch": 1, "enable_columnar_output": 1}
```

Comment:
This will fix `CodeQL` error.

```suggestion
        exec_args = {}
        if builder.has_groupby and not builder.has_join:
            exec_args = {"enable_lazy_fetch": 0, "enable_columnar_output": 0}
        elif not builder.has_groupby and builder.has_join:
            exec_args = {"enable_lazy_fetch": 1, "enable_columnar_output": 1}
```

Reply:
Yes, but it creates a new dict that is never used.

Reply:
it will be used in the case: `builder.has_groupby == builder.has_join:`

Yes, it is not used in all cases, but the impact on performance is very small, and the readability of the code is better.

Reply:
Ah, sorry, didn't notice you added the new condition.